components/org.wso2.carbon.identity.organization.user.invitation.management/src/main/java/org/wso2/carbon/identity/organization/user/invitation/management/InvitationCoreServiceImpl.java

```diff
@@ -0,0 +1,302 @@
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.ErrorMessage.ERROR_CODE_INVALID_CONFIRMATION_CODE;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.ErrorMessage.ERROR_CODE_INVALID_INVITATION_ID;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.ErrorMessage.ERROR_CODE_INVITATION_EXISTS;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.ErrorMessage.ERROR_CODE_INVITATION_EXPIRED;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.ErrorMessage.ERROR_CODE_NO_INVITATION_FOR_USER;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.ErrorMessage.ERROR_CODE_UNABLE_TO_RESEND_INVITATION;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.ErrorMessage.ERROR_CODE_UNSUPPORTED_FILTER_ATTRIBUTE;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.ErrorMessage.ERROR_CODE_UNSUPPORTED_FILTER_ATTRIBUTE_VALUE;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.ErrorMessage.ERROR_CODE_USER_NOT_FOUND;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.FILTER_STATUS_EQ;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.STATUS_EXPIRED;
+import static org.wso2.carbon.identity.organization.user.invitation.management.constant.UserInvitationMgtConstants.STATUS_PENDING;
+
+/**
+ * Implementation of the invitation core service which manages the invitations of the shared users.
+ */
+public class InvitationCoreServiceImpl implements InvitationCoreService {
+
+    private static final Log LOG = LogFactory.getLog(InvitationCoreServiceImpl.class);
+    private static final UserInvitationDAO userInvitationDAO = new UserInvitationDAOImpl();
+
+    @Override
+    public Invitation createInvitation(Invitation invitation) throws UserInvitationMgtException {
+
+        String organizationId = Utils.getOrganizationId();
+        OrganizationManager organizationManager = new OrganizationManagerImpl();
+        Invitation createdInvitation;
+        // ToDo : Introduce a new validation to check the invitation with the existing invitations.
+        try {
+            // Checking the user existence in the parent organization
+            List<String> ancestorsOrganizationsList =
+                    organizationManager.getAncestorOrganizationIds(organizationId);
+            // Picking the immediate parent organization id
```

Comment:
```suggestion
            // Picking the parent organization id
```
An organization has only one parent

Reply:
Done